lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Shared.hs

```diff
@@ -641,16 +643,36 @@ instance SupportsDiscovery n k => CompareDiscovery (SharedState n k) where
                     AddressPool.lookup addr (getPool intPool)
 
 instance Typeable n => KnownAddresses (SharedState n k) where
-    knownAddresses st = nonChangeAddresses
+    knownAddresses st =
+        nonChangeAddresses <> usedChangeAddresses <> pendingChangeAddresses
       where
-        -- TODO - After enabling txs for shared wallets we will need to expand this
         nonChangeAddresses = case ready st of
             Pending -> []
-            Active (SharedAddressPools extPool intPool _) ->
-                map swivel $ Map.toList $
-                AddressPool.addresses (getPool extPool) <>
+            Active (SharedAddressPools extPool _ _) ->
+                map (swivel utxoExternal) $ Map.toList $
+                AddressPool.addresses (getPool extPool)
+
+        swivel role' (k,(ix,s)) =
+            (liftPaymentAddress @n k, s, decoratePath st role' ix)
+
+        changeAddresses = case ready st of
+            Pending -> []
```

Comment:
Suggestion: Move all pattern matches on `ready st` to the outermost scope? I.e. all the `case` expressions can be removed if we make the pattern match at the function definition:

```
knownAddresses …{ready=Pending} = []
knownAddresses …{ready=Active (SharedAddressPools extPool intPool (PendingIxs ixs))} = 
  where
    …
```

Reply:
good idea - done